kubernetes/crd/domain-v1beta1-crd.yaml

```diff
@@ -31,6 +31,30 @@ spec:
           description: The specification of the operation of the WebLogic domain.
             Required.
           properties:
+            monitoringExporter:
+              type: object
+              description: Configuration for the use of the WebLogic Monitoring Exporter
+                as part of this domain.
+              properties:
+                image:
+                  type: string
+                  description: The WebLogic Monitoring Exporter sidecar image name.
+                    Defaults to oracle/weblogic-monitoring-exporter:2.0
+                imagePullPolicy:
+                  type: string
+                  description: The image pull policy for the WebLogic Monitoring Exporter
+                    sidecar image. Legal values are Always, Never, and IfNotPresent.
+                    Defaults to Always if image ends in :latest; IfNotPresent, otherwise.
+                  enum:
+                  - Always
+                  - Never
+                  - IfNotPresent
+                configuration:
+                  type: object
+                  description: The configuration for the WebLogic Monitoring Exporter
+                    sidecar. If specified, the operator will deploy a sidecar alongside
+                    each server instance. See https://github.com/oracle/weblogic-monitoring-exporter
+                  xKubernetesPreserveUnknownFields: true
```

Comment:
Isn't this supposed to be: `x-kubernetes-preserve-unknown-fields: true`

Reply:
In the code it is - and then Kubernetes changes it to camel case, which is confusing. I spent a good while trying to understand that. Recall that this is a generated file.

Reply:
I'm worried that this won't work. For the dedicated mode use case, customers create the CRD in advance by `kubectl create -f` with this document. This misnamed field likely won't do what it needs to. When the operator creates the CRD by directly creating the CRD does the CRD get the correct contents?

Reply:
Kubernetes is definitely getting the correct value when the operator generates the CRD this way - without this field in the CRD, it doesn't preserve the exporter configuration.

Reply:
You can simulate that test this way:
1. Uninstall the operator and delete the CRD
2. Install the CRD using `kubectl create -f <crd.yaml>`
3. Install the operator
4. (Optional) make sure that the operator doesn't overwrite the CRD -- it shouldn't because of the hash label.
5. See if Kubernetes drops the exporter configuration now.

Reply:
Or, you could just do the equivalent of a `sed` before writing the CRD to file. :)

Reply:
You're correct. If I create the CRD from this file, it ignores the annotation. If I edit the field to turn it back to dash case, it works. So I need to fix the generator to bypass whatever has been converting it.

Reply:
The problem is that the way the Yaml is written involves classes with maps of V1JsonSchemaProps objects, which contains fields like this
```
  public static final String SERIALIZED_NAME_X_KUBERNETES_PRESERVE_UNKNOWN_FIELDS = "x-kubernetes-preserve-unknown-fields";
  @SerializedName("x-kubernetes-preserve-unknown-fields")
  private Boolean xKubernetesPreserveUnknownFields;
```
When serialized by Gson, the serialized name is used - but Kubernetes uses SnakeYaml to write this, and it doesn't know about that annotation. So I do in fact need to process the lines as they are written, as though simulating `sed`. ::sigh::

Reply:
Alternately, I can change the YAML generation to generate _JSON_ and then convert that to YAML. That would probably be more obvious and safer.

Reply:
And that two-step process works, although it changes the order of some fields in the output.

Reply:
The changed order makes it a little hard to review, but I did my best... I think it looks good.